enhancements/cluster-logging/loki-tokenized-auth-enablement.md

```diff
@@ -0,0 +1,343 @@
+
+This enhancement seeks to bring a unified integration to configure LokiStack with CCO's capabilities to use tokenized authentication for accessing the respective Cloud Providers' object storage service. This improves the UX for consumers of the Loki Operator and in turn the Red Hat OpenShift Logging product.
+
+## Motivation
+
+As previously accomplished by other Red Hat and third-party operators (e.g. OADP Operator, External DNS Operator, AWS EFS CSI Driver Operator) the following proposal seeks to use the CCO's CredentialsRequest API in addition to the OpenShift Console OLM capabilities to detect STS-enabled clusters and manage the whole operator lifecycle.
+
+
+### User Stories
+
+* As a LokiStack administrator I want to off-load STS configuration to CCO so that I do not need to manually manage STS credentials configuration on AWS and in turn a custom LokiStack S3 object storage secret.
+* As a LokiStack administrator I want to off-load Azure WIF configuration to CCO so that I do not need to manually manage WIF credentials configuration on Azure and in turn a custom LokiStack Azure object storage secret.
+* As a LokiStack administrator I want to off-load GCP WIF configuration to CCO so that I do not need to manually manage WIF credentials configuration on GCP and in turn a custom LokiStack GCS object storage secret.
+
+### Goals
+
+Allow LokiStack administrators to manage LokiStack Cloud Provider related configuration (i.e. object storage service access) like with any other Red Hat Operator enabled for tokenized authentication.
+
+Allow OpenShift administrators to pre-provision Cloud Provider IAM resources for the layered product Red Hat OpenShift Logging and specifically the Loki Operator before installation.
+
+Allow Hosted Control Planes (HCP) and Red Hat OpenShift on AWS (ROSA) cluster administrators to to pre-provision Hosted Provider IAM resources for the layered product Red Hat OpenShift Logging and specifically the Loki Operator before installation.
+
+### Non-Goals
+
+Automating pre-provisioned Cloud Providers' IAM resources so that users are left only with the Loki Operator and the LokiStack resource installation only.
+
+## Proposal
+
+The Loki Operator supports per Cloud Provider the following annotations in it's respectives ClusterServiceVersion per bundle (community-openshift, openshift):
+
+- For AWS STS: `features.operators.openshift.io/token-auth-aws: true`
+- For Azure WIF: `features.operators.openshift.io/token-auth-azure: true`
+- For GCP WIF: `features.operators.openshift.io/token-auth-gcp: true`
+
+These will enable to operator installation to require specific IAM resource identifiers (e.g. role, client id, tenant id, etc.) which will be populated to the subscription and in turn exposed in the operator container environment.
```

Comment:
Some subtle grammar changes, but I would shorten this since it is long:

 These enable the operator installation to require specific IAM resource identifiers (e.g. role, client id, tenant id, etc.).  The operator subscription is populated with the identifiers and exposed in the operator container environment.